snorkel/models/candidate.py

```diff
@@ -80,6 +80,10 @@ def get_contexts(self):
         """Get a tuple of the consituent contexts making up this candidate"""
         return tuple(getattr(self, name) for name in self.__argnames__)
 
+    def get_parent(self):
```

Comment:
What I think we ideally want to do here (not now, but down the road) is return the lowest shared context in the context hierarchy.  For now though, perhaps we should either (a) explicitly check to make sure they have same parent, else throw error, or (b) explicitly name this `get_first_parent`?